src/compas_fab/robots/tool.py

```diff
@@ -145,3 +146,53 @@ def to_json(self, filepath):
         """
         with open(filepath, 'w') as f:
             json.dump(self.data, f, indent=4, sort_keys=True)
+
+    def from_tcf_to_t0cf(self, frames_tcf):
+        """Converts a list of frames at the robot's tool tip (tcf frame) to frames at the robot's flange (tool0 frame).
+
+        Parameters
+        ----------
+        frames_tcf : list of :class:`Frame`
+            Frames (in WCF) at the robot's tool tip (tcf).
+
+        Returns
+        -------
+        list of :class:`Frame`
+            Frames (in WCF) at the robot's flange (tool0).
+
+        Examples
+        --------
+        >>> mesh = Mesh.from_stl(compas_fab.get('planning_scene/cone.stl'))
+        >>> frame = Frame([0.14, 0, 0], [0, 1, 0], [0, 0, 1])
+        >>> tool = Tool(mesh, frame)
+        >>> frames_tcf = [Frame((-0.309, -0.046, -0.266), (0.276, 0.926, -0.256), (0.879, -0.136, 0.456))]
+        >>> tool.from_tcf_to_t0cf(frames_tcf)
+        [Frame(Point(-0.363, 0.003, -0.147), Vector(0.388, -0.351, -0.852), Vector(0.276, 0.926, -0.256))]
+        """
+        Te = Transformation.from_frame_to_frame(self.frame, Frame.worldXY())
+        return [Frame.from_transformation(Transformation.from_frame(f) * Te) for f in frames_tcf]
```

Comment:
From the doc string it seems like this could be an application of `Frame.local_to_local_coordinates`.  Is that true?

Reply:
No, unfortunately not.